XenModel/Utils/Helpers.cs

```diff
@@ -395,6 +395,25 @@ public static bool ElyOrGreater(Host host)
                 platform_version != null && productVersionCompare(platform_version, "2.1.1") >= 0;
         }
 
+        /// <param name="conn">May be null, in which case true is returned.</param>
+        public static bool HavanaOrGreaterLTSR71(IXenConnection conn)
+        {
+            return conn == null || HavanaOrGreaterLTSR71(Helpers.GetMaster(conn));
+        }
+
+         /// As Havana platform version is same with Ely and Honolulu, so use product version here
+        /// <param name="host">May be null, in which case true is returned.</param>
+        public static bool HavanaOrGreaterLTSR71(Host host)
+        {
+            if (host == null)
+                return true;
+             string product_version = HostProductVersion(host);
+            return
+                product_version != null && 
+                ((product_version.StartsWith("7.1") && productVersionCompare(product_version, "7.1.2") >= 0) || 

```

Comment:
Instead of hardcoding the product versions here, we should add them to the branding, similar to what we have for 7.0 for example https://github.com/xenserver/xenadmin/blob/master/Branding/branding.sh#L107